src/adapter.rs

```diff
@@ -45,6 +47,12 @@ impl Adapter {
         self.0.wait_available().await
     }
 
+    /// Check if adapter is available
+    #[inline]
+    pub fn is_available(&self) -> bool {
+        self.0.is_available()
+    }
```

Comment:
If this method is added, it needs to be implemented for all platforms. Perhaps it should be a separate PR?